packages/devtools_app/lib/src/debugger/flutter/debugger_controller.dart

```diff
@@ -370,26 +369,19 @@ class DebuggerController extends DisposableController
     });
     _sortedScripts.value = scriptRefs;
 
-    // ignore: unawaited_futures
-    (_service as VmServiceWrapper)
-        .getClassList(isolateRef.id)
-        .then((Response response) {
-      final List classes = response.json['classes'];
-      final List<ClassRef> classRefs = classes
-          .map((json) => ClassRef.parse(json))
-          .cast<ClassRef>()
+    try {
+      final classList = await _service.getClassList(isolateRef.id);
+      final classes = classList.classes
           .where((c) => c.name != null && c.name.isNotEmpty)
           .toList();
-
-      classRefs.sort((ClassRef a, ClassRef b) {
+      classes.sort((ClassRef a, ClassRef b) {
         // We sort uppercase so that items like Foo sort before items like _Foo.
         return a.name.toUpperCase().compareTo(b.name.toUpperCase());
       });
-
-      _sortedClasses.value = classRefs;
-    }).catchError((e) {
+      _sortedClasses.value = classes;
+    } catch (e) {
       // Fail gracefully - not all clients support getClassList().
```

Comment:
Is this actually the case? This RPC has been around for years but it just wasn't documented properly. Since it wasn't private, I doubt you'll run into a situation where this RPC isn't present.

Reply:
Ah - the web backends (`package:dwds`) won't have this call yet :)

Reply:
Of course... I always seem to forget that the VM service protocol isn't only implemented by the VM service :)

Reply:
we would run into the situation where the getClassList method isn't available from package:vm_service though (pre 4.0.2)